src/erc4626/ERC4626Oracle.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {IERC20} from "./IERC20.sol";
+import {IERC4626} from "./IERC4626.sol";
+
+contract ERC4626Oracle {
+    error OracleUnsupportedPair();
+
+    IERC4626 public immutable VAULT;
+    address public immutable ASSET;
```

Comment:
Even if `asset` in ERC4626 returns an address, [it is required to be an ERC20](https://github.com/ethereum/ercs/blob/master/ERCS/erc-4626.md#asset), so here you can use IERC20 as the type for `ASSET`, which would be cleaner.